ql/src/java/org/apache/hadoop/hive/ql/parse/TaskCompiler.java

```diff
@@ -517,17 +517,21 @@ private Path getDefaultCtasLocation(final ParseContext pCtx) throws SemanticExce
     try {
       String protoName = null, suffix = "";
       boolean isExternal = false;
-      
+      boolean useSuffix = HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_CREATE_TABLE_USE_SUFFIX)
+              || HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED);
+
       if (pCtx.getQueryProperties().isCTAS()) {
         protoName = pCtx.getCreateTable().getDbTableName();
         isExternal = pCtx.getCreateTable().isExternal();
-      
+        if (!isExternal && useSuffix) {
+          long txnId = Optional.ofNullable(pCtx.getContext())
```

Comment:
could we extract this part into the helper method, it's repeating in multiple places

Reply:
Updated and added a new function called "getTableOrMVSuffix"